addon/components/modal-component/component.js

```diff
@@ -0,0 +1,21 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  classNames: ['modal', 'fade'],
+
+  _: Ember.observer('open', function() {
+    const mode =  this.get('open') === true ? 'show' : 'hide';
+    this.$().modal(mode);
```

Comment:
instead of keeping a state in this component, can't we use `.modal('toggle')` cf. http://v4-alpha.getbootstrap.com/components/modal/#modaltoggle